query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/QueryRequestToPinotSQLConverter.java

```diff
@@ -200,43 +199,18 @@ private Expression handleValueConversionForLiteralExpression(Expression lhs, Exp
       return rhs;
     }
 
-    Expression newRhs = rhs;
     String lhsColumnName = lhs.getColumnIdentifier().getColumnName();
-
-    ToValueConverter converter = getValueConverter(rhs.getLiteral().getValue().getValueType());
```

Comment:
@kotharironak I know you have added this. The requirement I see here is mainly around doing specific transformation on the destination column type in Pinot, not about what was the type in the incoming request. Hence, I thought we don't really need an interface since we'll always have only one implementation.